### PR TITLE
chore: update Angular v17

### PR DESCRIPTION
**Pull request by MunMunMiao**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[x] Other... Please describe:
```

## What is the current behavior?

Issue Number: #8142

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by MunMunMiao** [comment]
test error maybe need support standalone #8095

File: .eslintrc.js [hunk 1]
```diff
@@ -31,7 +31,6 @@ module.exports = {
           ],
           2
         ],
-        'jsdoc/newline-after-description': 1,
```
Comment (MunMunMiao) [inline .eslintrc.js:34]: Because 'jsdoc/new after description' is obsolete, we need to remove it

**Comment by Nicoss54** [comment]
@MunMunMiao i think the dependencies ngx-qwicklink and ant-design/angular-icons should be updated when it will be released.  Alors because test use customwebpack/buiders package this dependencies need to be update too when it will be released

**Comment by MunMunMiao** [comment]
> @MunMunMiao i think the dependencies ngx-qwicklink and ant-design/angular-icons should be updated when it will be released. Alors because test use customwebpack/buiders package this dependencies need to be update too when it will be released

yes. i think we need waitting ```ngx-qwicklink``` and ```ant-design/angular-icons``` and ```@angular-builders/custom-webpack```

**Comment by Nicoss54** [comment]
> > @MunMunMiao i think the dependencies ngx-qwicklink and ant-design/angular-icons should be updated when it will be released. Alors because test use customwebpack/buiders package this dependencies need to be update too when it will be released
> 
> yes. i think we need waitting `ngx-qwicklink` and `ant-design/angular-icons` and `@angular-builders/custom-webpack`

Mmh il will try to update ngx-qwicklink today

**Comment by MunMunMiao** [comment]
> > > @MunMunMiao i think the dependencies ngx-qwicklink and ant-design/angular-icons should be updated when it will be released. Alors because test use customwebpack/buiders package this dependencies need to be update too when it will be released
> > 
> > 
> > yes. i think we need waitting `ngx-qwicklink` and `ant-design/angular-icons` and `@angular-builders/custom-webpack`
> 
> Mmh il will try to update ngx-qwicklink today

I have submitted a PR :)

**Comment by Nicoss54** [comment]
> > > > @MunMunMiao i think the dependencies ngx-qwicklink and ant-design/angular-icons should be updated when it will be released. Alors because test use customwebpack/buiders package this dependencies need to be update too when it will be released
> > > 
> > > 
> > > yes. i think we need waitting `ngx-qwicklink` and `ant-design/angular-icons` and `@angular-builders/custom-webpack`
> > 
> > 
> > Mmh il will try to update ngx-qwicklink today
> 
> I have submitted a PR :)

Awesome work ! Thanks a lot

**Comment by MarcosPaulo775** [comment]
Does the new build support? (`@angular-devkit/build-angular:browser-esbuild` and `@angular-devkit/build-angular:application`)

**Comment by nguyendkn** [comment]
I'm looking forward

**Comment by MunMunMiao** [comment]
> Does the new build support? (`@angular-devkit/build-angular:browser-esbuild` and `@angular-devkit/build-angular:application`)

We currently do not plan to support it. But in the future, I will try to remove webpack

**Comment by Ninja4Panda** [comment]
can't wait 😀

File: package.json [hunk 2]
```diff
@@ -25,41 +25,40 @@
...
   },
   "dependencies": {
-    "@angular/cdk": "^16.2.1",
+    "@angular/cdk": "^17.0.0",
     "@ant-design/icons-angular": "^16.0.0",
     "date-fns": "^2.16.1",
     "ngx-quicklink": "^0.4.2"
   },
   "devDependencies": {
     "@angular-builders/custom-webpack": "16.0.1",
```
Comment (markpritchett) [inline package.json:34]: https://www.npmjs.com/package/@angular-builders/custom-webpack/v/17.0.0 now released

File: package.json [hunk 3]
```diff
@@ -25,41 +25,40 @@
     "prepare": "husky install"
   },
   "dependencies": {
-    "@angular/cdk": "^16.2.1",
+    "@angular/cdk": "^17.0.0",
     "@ant-design/icons-angular": "^16.0.0",
```
Comment (markpritchett) [inline package.json:29]: https://www.npmjs.com/package/@ant-design/icons-angular/v/17.0.0 released

**Comment by MunMunMiao** [comment]
now ```custom webpack``` and ```icons angular``` have been upgraded to v17

**Comment by HyperLife1119** [comment]
是否有任何方案可以让 zorro 摆脱 custom webpack ?

**Comment by simplejason** [comment]
> 是否有任何方案可以让 zorro 摆脱 custom webpack ?

目前这用在了测试上，因为支持了 antd v4 less 变量 root-entry-name，我们后面需要探索是否可以使用自带的 fileReplacements 来实现脱离 custom webpack

File: components/package.json [hunk 4]
```diff
@@ -27,17 +27,17 @@
     "url": "https://github.com/NG-ZORRO/ng-zorro-antd/issues"
   },
   "dependencies": {
-    "@angular/cdk": "^16.0.0",
+    "@angular/cdk": "^17.0.0",
     "@ant-design/icons-angular": "^16.0.0",
```
Comment (jornare) [inline components/package.json:31]: forgot this one to 17.. :)

Comment (MunMunMiao) [reply components/package.json]: > forgot this one to 17.. :)

thank you, now it has been fixed

**Pull request merged**